cli/azd/pkg/azure/arm_template.go

```diff
@@ -20,6 +28,22 @@ type ArmTemplate struct {
 	Outputs        ArmTemplateOutputs              `json:"outputs"`
 }
 
+func (t ArmTemplate) TargetScope() DeploymentScope {
```

Comment:
Can you add docs here? It's not clear what are the expectations... 

Seems like depending on the name of the template that we are swapping between deployment vs subscription?

Reply:
Will add a comment here.

The compiled arm template has a `$schema` property and that value varies depending on if you are targeting [resource group](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-to-resource-group?tabs=azure-cli#schema) or [subscription scope](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-to-subscription?tabs=azure-cli#schema).

We exploit this to determine the author's intended scope.

Reply:
Added some docs.